### DISCUSSION TITLE
How to change uid/gid for uploaded files via FTP

### DISCUSSION THREAD

**Discussion by ruzickap** [drakkan/sftpgo#1147] [Q&A] [answered]
I'm using MacOS and SFTPGo installed using brew.
Then I enabled FTP service in SFTPGo and created single user.

Commands:

```bash
brew install sftpgo

cat > /tmp/sftpgo-initprovider-data.json << EOF
{
  "users": [
    {
      "id": 1,
      "status": 1,
      "username": "test",
      "password": "xxxxxxxx",
      "home_dir": "/Users/ruzickap/Pictures/ftp",
      "permissions": {
        "/": [
          "*"
        ]
      }
    }
  ],
  "folders": [],
  "admins": [
    {
      "id": 1,
      "status": 1,
      "username": "admin",
      "password": "xxxxxxxx",
      "email": "",
      "permissions": [
        "*"
      ],
      "filters": {}
    }
  ],
  "version": 7
}
EOF

sftpgo initprovider --config-dir /usr/local/var/sftpgo --loaddata-from /tmp/sftpgo-initprovider-data.json

cat > /usr/local/etc/sftpgo/sftpgo.json << EOF
{
  "ftpd": {
    "bindings": [
      {
        "port": 21
      }
    ]
  },
  "sftpd": {
    "bindings": [
      {
        "port": 0
      }
    ]
  }
}
EOF

sudo brew services restart sftpgo
```

When I upload some file using FTP they are always owned by `root` / `staff`.

Is it docmented somewhere how to change the user owner to something else than `root` like it is mentioned here: https://github.com/drakkan/sftpgo/blob/8be8343fee174efbc4f64f406ef3fc6079ace59b/README.md?plain=1#L83

(I was not able to find the details...)

Thank you...

**Comment by drakkan** [comment, marked as answer]
set uid/gid from the edit user page

![Schermata del 2023-01-15 17-32-37](https://user-images.githubusercontent.com/553263/212553606-4353282b-d6d1-4df7-9c1b-c565ad9a55c1.png)

or using the REST API

**Reply by ruzickap** [reply]
Oh... that was simple... 
You are right :-)

Thanks for quick reaction...

**Comment by Eliophot** [comment]
Hello, set uid/gid from the edit user page not present in 2.7... I try with rest api, it's ok.

**Reply by CoreKit-glitch** [reply]
Correct, same behavior on my side.
